extensions/replication/utils/SetupReplication.js

```diff
@@ -103,15 +98,31 @@ class SetupReplication {
      * @param {RoundRobin} params.target.hosts - destination hosts
      * @param {String} [params.target.transport] - transport protocol for
      *   target (http/https)
+     * @param {Boolean} [params.checkSanity=false] - whether to check
+     *   sanity of the config after setup, in case something would
+     *   have gone wrong but unnoticed
+     * @param {Number} [params.retryTimeoutS=300] - timeout for
+     *   request retries
+     * @param {Boolean} [params.skipSourceBucketCreation=false] - can
+     *   be set to true if the source bucket is guaranteed to exist to
+     *   spare a request
      * @param {Object} params.log - werelogs request logger object
      */
     constructor(params) {
-        const { source, target, log } = params;
+        const { source, target, checkSanity,
+                retryTimeoutS, skipSourceBucketCreation, log } = params;
+        super({ retryTimeoutS });
         this._log = log;
         this._sourceBucket = source.bucket;
         this._targetBucket = target.bucket;
+        this._checkSanityEnabled = checkSanity || false;
+        this._skipSourceBucketCreation = skipSourceBucketCreation || false;
         this.destHosts = target.hosts;
         const destHost = this.destHosts.pickHost();
+        // XXX use target port through nginx gateway
+        // Hard-code port 8600 for now on IAM client while other vault
+        // clients go through the gateway (signature mismatch issue).
+        const destPort = (target.vault && target.vault.adminPort) || 8600;
```

Comment:
I think it works with javascript behavior of `||`, correct me if I'm wrong: when `target.vault.adminPort` is set (and not 0 or anything falsy), the expression will take its value and the part after `||` is not considered, but if it's not set or falsy, 8600 is the expression value instead, which matches what we want here, since port numbers cannot be 0 it should be good.

Reply:
ah ya you're right, it does set. Sorry I just assumed the && would return a boolean instead of the intended adminPort